FastBank.Services/MessageService/MessageService.cs

```diff
@@ -75,19 +70,156 @@ public Message InputMessage(User user)
             return message;
         }
 
-        public void ShowMessagesMenu(User user)
+        public Message ReplayToMessage(User user, Message message)
+        {
+            Console.WriteLine("Please intput replay to text message");
+            Console.Write("Replay: ");
+            var text = Console.ReadLine() ?? string.Empty;
+
+            var replayMessage = new Message(
+                                        Guid.NewGuid(), 
+                                        user, 
+                                        message.Sender, 
+                                        Roles.Customer,
+                                        text, 
+                                        $"Re: {message.Subject}",
+                                        message, 
+                                        MessageStatuses.Sent, 
+                                        MessageType.Inquery, 
+                                        null);
+
+            _messageRepo.Add(replayMessage);
+            _messageRepo.UpdateStatus(message, MessageStatuses.Replied);
+
+            return replayMessage;
+        }
+
+        public void ShowMessagesMenu(User user, List<Message?>? messages = null)
```

Comment:
Why do you expect to have List<Message?>?

Reply:
![image](https://github.com/bulgarchik/FastBank/assets/41870483/04a6f845-0b33-4b90-8252-561876771ab2)
On opening Customer Service menu I am getting messages to notify user about his new messages. So to reduce calls to db put it like argument to method. And that is why this method expect to have List<Message>

Reply:
Ok, maybe it is good to review this